docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+   Extending the immutability period at the bucket level to cover the extended hibernation period might seem like a solution. However, this approach is **not feasible** due to several significant drawbacks:
+
+   - **Global Impact on All Objects:** Since bucket-level immutability settings apply to all objects within the bucket, extending the immutability period would affect not only the existing snapshots but also all future snapshots uploaded to the bucket.
+   - **Increased Storage Costs:** Prolonging the immutability period means that snapshots cannot be deleted or modified until the extended period expires. This leads to the accumulation of old backups, consuming more storage space and significantly driving up storage costs over time.
+   - **Inefficient Resource Utilization:** Retaining all snapshots for a longer period than necessary is an inefficient use of storage resources, especially when only the latest snapshots are typically needed for restoration purposes.
+
+   Given these considerable disadvantages, extending the bucket-level immutability period is impractical for maintaining the immutability of snapshots during extended hibernation periods.
+
+2. **Take a New Snapshot**
+
+   By creating a new snapshot, immutability is retained for the new snapshot without affecting the immutability settings of the entire bucket or other snapshots.
+
+   Approach 2 is feasible and can be achieved in two ways:
+
+   - **Option A:** Bring the ETCD cluster back up by increasing the replicas, take a snapshot, and then hibernate the ETCD by setting the replicas back to zero **without the operator's intention**.
+     - **Challenges:**
+       - **Unintended State Changes:** Increasing the ETCD replicas to bring the cluster back up modifies the cluster's state without the operator's explicit intention, which could lead to unexpected behavior or conflicts with operational policies.
+       - **Operational Overhead:** Automating the process of scaling the ETCD cluster up and down adds complexity and potential risks, especially if the scaling operations interfere with other scheduled tasks or maintenance windows.
+
+   - **Option B:** Start an embedded ETCD instance and take a snapshot.
+     - **Advantages:**
+       - **Respects Operator Intentions:** Does not alter the state of the actual ETCD cluster, keeping it in hibernation as intended by the operator.
+       - **No Impact on Existing Setup:** Avoids modifying the ETCD cluster's replicas, preventing any unintended consequences.
+       - **Automated Process:** Can be integrated into existing maintenance jobs without manual intervention.
+
+**Proposed Solution:**
+
+*Option B is recommended* because it respects the operator's intention to keep the ETCD cluster hibernated and avoids the complexities and risks associated with modifying the cluster's state.
+
+Leverage the compaction job to take fresh snapshots periodically during hibernation. This ensures that:
```

Comment:
We are introducing compaction job here for the first time, so may a reference or link to understand what is a compaction job.

Also, I was wondering aren't we overloading the compaction job here, should be create a new job which deploys compaction controller code path but is specific for taking snapshots for hibernated clusters. It will also help in differentiation of errors with regular compaction job.